fonz/connection.py

```diff
@@ -0,0 +1,172 @@
+from typing import Sequence, List, Dict, Any, Optional
+from fonz.utils import compose_url
+import requests
+import logging
+import sys
+
+JsonDict = Dict[str, Any]
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(message)s')
+
+
+class Fonz:
+
+    def __init__(self, url: str, client_id: str, client_secret: str,
+                 port: int, api: str, project: str = None, branch: str = None):
+        """Instantiate Fonz and save authentication details and branch."""
+        if url[-1] == '/':
+            self.url = '{}:{}/api/{}/'.format(url[:-1], port, api)
+        else:
+            self.url = '{}:{}/api/{}/'.format(url, port, api)
```

Comment:
See [comments](https://github.com/dbanalyticsco/Fonz/pull/5/files?file-filters%5B%5D=.py&file-filters%5B%5D=.txt#diff-74d73df094675fff26fe99ef44260b6aR23) on `utils.py`. I'm guessing we can solve this with some `urllib` or `pathlib` functions.